src/window/window_wrapper/mod.rs

```diff
@@ -339,10 +341,32 @@ pub fn create_window(
             previous_frame_start = frame_start;
         }
 
+        if !was_centered {
+            let window = window_wrapper.windowed_context.window();
+            let new_size = window.inner_size();
+            if let Some(handle) = window.current_monitor() {
+                if is_already_resized(new_size) {
+                    center_window(handle, window);
+                    was_centered = true;
+                }
+            }
+        }
+
         *control_flow = ControlFlow::WaitUntil(previous_frame_start + frame_duration)
     });
 }
 
 fn is_already_resized(size: PhysicalSize<u32>) -> bool {
     size != PhysicalSize::from((800, 600))
 }
+
+fn center_window(handle: winit::monitor::MonitorHandle, window: &Window) {
+    let outer_size = window.outer_size();
+    let monitor_size = handle.size();
+    let window_x = (monitor_size.width as i32 - outer_size.width as i32) / 2;
+    let window_y = (monitor_size.height as i32 - outer_size.height as i32) / 2;
```

Comment:
Can this be `(monitor_size.width - outer_size.width) as i32 / 2`?
Should look a bit cleaner.

Reply:
I added the `as i32` afterwards, because both `monitor_size.width` and `outer_size.width` are usize, so the substraction would fail if `outer_size.width` is larger than `monitor_size.width`.

With the `as i32`, it does not fail and the window is centered.

Although I agree that it is not satisfying to have it 4 times in 2 lines, so don't hesitate if you can think of a better implementation :).

Reply:
Ah, they're usize. That changes things.
I'm not sure what the expected behaviour should be when `outer_size` is larger than `monitor_size`.
Maybe [saturating_sub](https://doc.rust-lang.org/stable/std/primitive.usize.html#method.saturating_sub) could be used? So then if `outer_size` is larger than `monitor_size` it will "snap" to top-left corner (I think).

Reply:
Maximizing it seems reasonable

Reply:
It should be better now